src/NServiceBus.TransactionalSession.AcceptanceTests.SqlP/When_not_using_outbox.cs

```diff
@@ -0,0 +1,108 @@
+﻿namespace NServiceBus.TransactionalSession.AcceptanceTests
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using AcceptanceTesting;
+    using NUnit.Framework;
+
+    [ExecuteOnlyForEnvironmentWith(EnvironmentVariables.SqlServerConnectionString)]
+    public class When_not_using_outbox : NServiceBusAcceptanceTest
+    {
+        [OneTimeSetUp]
+        public void Setup() => SqlSetup.Setup();
+
+        [Test]
+        public async Task Should_send_messages_on_transactional_session_commit()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<AnEndpoint>(s => s.When(async (_, ctx) =>
+                {
+                    using var scope = ctx.ServiceProvider.CreateScope();
+                    using var transactionalSession = scope.ServiceProvider.GetRequiredService<ITransactionalSession>();
+
+                    await transactionalSession.OpenSqlSession();
+
+                    await transactionalSession.SendLocal(new SampleMessage());
+
+                    await transactionalSession.Commit();
+                }))
+                .Done(c => c.MessageReceived)
+                .Run();
+        }
+
+        [Test]
+        public async Task Should_not_send_messages_if_session_is_not_committed()
+        {
+            var result = await Scenario.Define<Context>()
+                .WithEndpoint<AnEndpoint>(s => s.When(async (messageSession, ctx) =>
+                {
+                    using (var scope = ctx.ServiceProvider.CreateScope())
+                    using (var transactionalSession = scope.ServiceProvider.GetRequiredService<ITransactionalSession>())
+                    {
+                        await transactionalSession.OpenSqlSession();
```

Comment:
Thoughts for discussion or another PR:

It seems the only difference for all the tests is the concrete option method. This could be moved into a partial and then we could reuse the same tests for all persisters. Thoughts?

Reply:
Would be awesome